### PR TITLE
Adding sliding window attention support in cuDNN SDPA API

### PR DESCRIPTION
**Pull request by Cjkkkk**

### CODE REVIEW

**Comment by Cjkkkk** [comment]
@superbobry @kaixih could you guys help take a look?

**Comment by kaixih** [comment]
LGTM.

File: jax/_src/cudnn/fused_attention_stablehlo.py [hunk 1]
```diff
@@ -985,6 +1003,7 @@ def dot_product_attention(query: Array,
     dropout_rate: Dropout rate.
     qkv_layout: Layout string, with supported formats being BTNH, BNTH, BSNH,
                 BNSH.
+    sliding_window_length: Sliding window attention length, default disabled.
```
Comment (superbobry) [inline jax/_src/cudnn/fused_attention_stablehlo.py:1006]: Would it be obvious to users what sliding window is referring to? I'm not familiar with the underlying CuDNN API, so I'm personally not sure what the window is. Perhaps, add more context in the docstring?

File: jax/_src/cudnn/fused_attention_stablehlo.py [hunk 2]
```diff
@@ -953,7 +970,8 @@ def dot_product_attention(query: Array,
                           mask_type: MaskType = MaskType.NO_MASK,
                           seed: int = 42,
                           dropout_rate: float = 0.,
-                          qkv_layout: str = "BTNH"):
+                          qkv_layout: str = "BTNH",
+                          sliding_window_length: int = 0):
```
Comment (superbobry) [inline jax/_src/cudnn/fused_attention_stablehlo.py:974]: Does it make sense to use `int | None` to indicate that windowing is optional?

File: tests/fused_attention_stablehlo_test.py [hunk 3]
```diff
@@ -108,6 +121,8 @@ def get_encoded_padding_mask(encoded):
     bias = get_causal_mask(logits)
   elif mask_type == MaskType.PADDING:
     bias = get_padding_mask(logits)
+  elif sliding_window_length > 0:
```
Comment (superbobry) [inline tests/fused_attention_stablehlo_test.py:124]: Maybe fail if it's <0?

**Comment by Cjkkkk** [comment]
@superbobry all comments addressed, could you take another look? Thanks!

File: jax/_src/cudnn/fused_attention_stablehlo.py [hunk 4]
```diff
@@ -151,6 +152,7 @@ def create_dot_product_attention_backend_config(batch,
     "seed": seed,
     "is_flash_attention": True,
     "mask_type": convert_mask_type_to_string(mask_type),
+    "sliding_window_length": sliding_window_length if not None else 0,
```
Comment (kaixih) [inline jax/_src/cudnn/fused_attention_stablehlo.py:155]: Are we sure this is correct? `not None` is always true.

Comment (Cjkkkk) [reply jax/_src/cudnn/fused_attention_stablehlo.py:155]: It can be None, I just changed the sdpa to use None as default value.

Comment (Cjkkkk) [reply jax/_src/cudnn/fused_attention_stablehlo.py:155]: fixed.

**Review by superbobry — APPROVED** [review]

**Pull request merged**